src/Zicht/Bundle/SolrBundle/Command/ReindexCommand.php

```diff
@@ -59,68 +66,97 @@ protected function configure()
     /**
      * @{inheritDoc}
      */
+    protected function initialize(InputInterface $input, OutputInterface $output)
+    {
+        $this->entities = $input->getArgument('entity');
+        if (empty($this->entities)) {
+            $solr = $this->solrManager;
+            $metaCollection = $this
+                ->doctrine
+                ->getManager($input->getOption('em'))
+                ->getMetadataFactory()
+                ->getAllMetadata();
+            /** @var \Doctrine\ORM\Mapping\ClassMetadata $meta */
+            foreach ($metaCollection as $meta) {
```

Comment:
kan dit niet makkelijker? in de services zijn alle mappers al geconfigureerd met een tag die je kan ophalen op hun tag en kan bevragen voor welke entitiy ze zijn aangemaakt. in de AbstractDataMapper heb je de property $classNames. Daar staat deze informatie al in. daar moet misschien getClassNames() als functie ingebouwd worden en heel doctrine heb je hier dan niet zo nodig.

Reply:
Ideeen overgenomen in https://github.com/zicht/solr-bundle/pull/35